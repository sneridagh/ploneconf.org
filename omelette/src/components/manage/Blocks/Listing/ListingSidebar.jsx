import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Segment, Accordion } from 'semantic-ui-react';
import { defineMessages, FormattedMessage, injectIntl } from 'react-intl';

import {
  ListingBlockData as ListingData,
  ListingBlockStyle as ListingStyle,
  ListingBlockMore as ListingMore,
  Icon,
} from '@plone/volto/components';

import upSVG from '@plone/volto/icons/up-key.svg';
import downSVG from '@plone/volto/icons/down-key.svg';

const messages = defineMessages({
  blockStyle: {
    id: 'Block style',
    defaultMessage: 'Block style',
  },
  content: {
    id: 'Content',
    defaultMessage: 'Content',
  },
  more: {
    id: 'Link more',
    defaultMessage: 'Link more',
  },
});

const ListingSidebar = (props) => {
  const [activeAccIndex, setActiveAccIndex] = useState(1);

  function handleAccClick(e, titleProps) {
    const { index } = titleProps;
    const newIndex = activeAccIndex === index ? -1 : index;

    setActiveAccIndex(newIndex);
  }

  return (
    <Segment.Group raised>
      <header className="header pulled">
        <h2>
          <FormattedMessage id="Listing" defaultMessage="Listing" />
        </h2>
      </header>

      <Accordion fluid styled className="form">
        <ListingStyle {...props} />

        <Accordion.Title
          active={activeAccIndex === 1}
          index={1}
          onClick={handleAccClick}
        >
          {props.intl.formatMessage(messages.content)}
          {activeAccIndex === 1 ? (
            <Icon name={upSVG} size="20px" />
          ) : (
            <Icon name={downSVG} size="20px" />
          )}
        </Accordion.Title>
        <Accordion.Content active={activeAccIndex === 1}>
          <ListingData {...props} />
        </Accordion.Content>

        <Accordion.Title
          active={activeAccIndex === 2}
          index={2}
          onClick={handleAccClick}
        >
          {props.intl.formatMessage(messages.more)}
          {activeAccIndex === 2 ? (
            <Icon name={upSVG} size="20px" />
          ) : (
            <Icon name={downSVG} size="20px" />
          )}
        </Accordion.Title>
        <Accordion.Content active={activeAccIndex === 2}>
          <ListingMore {...props} />
        </Accordion.Content>
      </Accordion>
    </Segment.Group>
  );
};

ListingSidebar.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
  block: PropTypes.string.isRequired,
  onChangeBlock: PropTypes.func.isRequired,
};

export default injectIntl(ListingSidebar);
