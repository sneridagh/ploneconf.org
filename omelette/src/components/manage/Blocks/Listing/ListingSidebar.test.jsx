import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import { render, waitFor } from '@testing-library/react';

import ListingSidebar from './ListingSidebar';

const mockStore = configureStore();

test('renders a Listing Block Sidebar component', async () => {
  const store = mockStore({
    querystring: { sortable_indexes: {} },
    intl: {
      locale: 'en',
      messages: {},
    },
  });
  const { container } = render(
    <Provider store={store}>
      <ListingSidebar
        id="dcdf1f42-645d-48f6-9531-357bdc2e1881"
        data={{
          '@type': 'listing',
          query: [
            {
              i: 'review_state',
              o: 'plone.app.querystring.operation.selection.any',
              v: ['private'],
            },
          ],
        }}
        block="1234"
        pathname="/news"
        onChangeBlock={() => {}}
      />
    </Provider>,
  );
  await waitFor(() => {});
  expect(container.firstChild).toMatchSnapshot();
});
