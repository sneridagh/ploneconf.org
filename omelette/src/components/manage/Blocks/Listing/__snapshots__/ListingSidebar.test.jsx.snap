// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Listing Block Sidebar component 1`] = `
<div
  class="ui raised segments"
>
  <header
    class="header pulled"
  >
    <h2>
      Listing
    </h2>
  </header>
  <div
    class="accordion ui fluid styled form"
  >
    <div
      class="sidebar-listing-data listing-style"
    >
      <div
        class="inline required field"
        id="field-template"
      >
        <div
          class="ui grid"
        >
          <div
            class="stretched row"
          >
            <div
              class="four wide column"
            >
              <div
                class="wrapper"
              >
                <label
                  for="select-listingblock-template"
                >
                  Template
                </label>
              </div>
            </div>
            <div
              class="eight wide column"
            >
              <div
                class="react-select-container css-2b097c-container"
                id="select-listingblock-template"
              >
                <div
                  class="react-select__control css-1ard45z-control"
                >
                  <div
                    class="react-select__value-container react-select__value-container--has-value css-1a9cv00-ValueContainer"
                  >
                    <div
                      class="react-select__single-value css-1uccc91-singleValue"
                    >
                      Default
                    </div>
                    <div
                      class="css-b8ldur-Input"
                    >
                      <div
                        class="react-select__input"
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="react-select-2-input"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class="react-select__indicator-separator css-6fp5c8-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="react-select__indicator react-select__dropdown-indicator css-7hdh9a-DropdownIndicator"
                    >
                      <svg
                        class="icon"
                        style="height: 24px; width: auto; fill: #007bc1;"
                        viewBox=""
                        xmlns=""
                      />
                    </div>
                  </div>
                </div>
                <input
                  name="select-listingblock-template"
                  type="hidden"
                  value="default"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="active title"
    >
      Content
      <svg
        class="icon"
        style="height: 20px; width: auto; fill: currentColor;"
        viewBox=""
        xmlns=""
      />
    </div>
    <div
      class="content active"
    >
      <div
        class="ui segment form sidebar-listing-data"
      >
        <div
          class="inline field"
          id="field-source"
        >
          <div
            class="ui grid"
          >
            <div
              class="stretched row"
            >
              <div
                class="twelve wide column"
              >
                <div
                  class="simple-field-name"
                >
                  Criteria
                </div>
              </div>
            </div>
            <div
              class="stretched row"
            >
              <div
                class="twelve wide column"
              >
                <div
                  class="fields"
                >
                  <div
                    class="field"
                    style="flex: 1 0 auto;"
                  >
                    <div
                      class="react-select-container css-2b097c-container"
                      id="field-source"
                    >
                      <div
                        class="react-select__control css-1ard45z-control"
                      >
                        <div
                          class="react-select__value-container css-1a9cv00-ValueContainer"
                        >
                          <div
                            class="react-select__placeholder css-1wa3eu0-placeholder"
                          >
                            Add criteria
                          </div>
                          <div
                            class="css-b8ldur-Input"
                          >
                            <div
                              class="react-select__input"
                              style="display: inline-block;"
                            >
                              <input
                                aria-autocomplete="list"
                                autocapitalize="none"
                                autocomplete="off"
                                autocorrect="off"
                                id="react-select-3-input"
                                spellcheck="false"
                                style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                                tabindex="0"
                                type="text"
                                value=""
                              />
                              <div
                                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                        >
                          <span
                            class="react-select__indicator-separator css-6fp5c8-indicatorSeparator"
                          />
                          <div
                            aria-hidden="true"
                            class="react-select__indicator react-select__dropdown-indicator css-7hdh9a-DropdownIndicator"
                          >
                            <svg
                              class="icon"
                              style="height: 24px; width: auto; fill: #007bc1;"
                              viewBox=""
                              xmlns=""
                            />
                          </div>
                        </div>
                      </div>
                      <input
                        name="source"
                        type="hidden"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="inline field"
          id="field-listingblock-sort-on"
        >
          <div
            class="ui grid"
          >
            <div
              class="stretched row"
            >
              <div
                class="four wide column"
              >
                <div
                  class="wrapper"
                >
                  <label
                    for="select-listingblock-sort-on"
                  >
                    Sort on
                  </label>
                </div>
              </div>
              <div
                class="eight wide column"
              >
                <div
                  class="react-select-container css-2b097c-container"
                  id="select-listingblock-sort-on"
                >
                  <div
                    class="react-select__control css-1ard45z-control"
                  >
                    <div
                      class="react-select__value-container react-select__value-container--has-value css-1a9cv00-ValueContainer"
                    >
                      <div
                        class="react-select__single-value css-1uccc91-singleValue"
                      >
                        No selection
                      </div>
                      <div
                        class="css-b8ldur-Input"
                      >
                        <div
                          class="react-select__input"
                          style="display: inline-block;"
                        >
                          <input
                            aria-autocomplete="list"
                            autocapitalize="none"
                            autocomplete="off"
                            autocorrect="off"
                            id="react-select-4-input"
                            spellcheck="false"
                            style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                            tabindex="0"
                            type="text"
                            value=""
                          />
                          <div
                            style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class="react-select__indicators css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class="react-select__indicator-separator css-6fp5c8-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class="react-select__indicator react-select__dropdown-indicator css-7hdh9a-DropdownIndicator"
                      >
                        <svg
                          class="icon"
                          style="height: 24px; width: auto; fill: #007bc1;"
                          viewBox=""
                          xmlns=""
                        />
                      </div>
                    </div>
                  </div>
                  <input
                    name="select-listingblock-sort-on"
                    type="hidden"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="inline field"
        >
          <div
            class="ui grid"
          >
            <div
              class="stretched row"
            >
              <div
                class="twelve wide column"
              >
                <div
                  class="wrapper"
                >
                  <div
                    class="ui checkbox"
                  >
                    <input
                      class="hidden"
                      name="field-listingblock-sort-on-reverse"
                      readonly=""
                      tabindex="0"
                      type="checkbox"
                      value=""
                    />
                    <label
                      for="field-listingblock-sort-on-reverse"
                    >
                      Reversed order
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="inline field text"
        >
          <div
            class="ui grid"
          >
            <div
              class="stretched row"
            >
              <div
                class="four wide column"
              >
                <div
                  class="wrapper"
                >
                  <label
                    for="field-limit"
                  >
                    Results limit
                  </label>
                </div>
              </div>
              <div
                class="eight wide column"
              >
                <div
                  class="ui input"
                >
                  <input
                    id="field-limit"
                    name="limit"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="inline field text"
        >
          <div
            class="ui grid"
          >
            <div
              class="stretched row"
            >
              <div
                class="four wide column"
              >
                <div
                  class="wrapper"
                >
                  <label
                    for="field-itembatchsize"
                  >
                    Item batch size
                  </label>
                </div>
              </div>
              <div
                class="eight wide column"
              >
                <div
                  class="ui input"
                >
                  <input
                    id="field-itembatchsize"
                    name="itembatchsize"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="title"
    >
      Link more
      <svg
        class="icon"
        style="height: 20px; width: auto; fill: currentColor;"
        viewBox=""
        xmlns=""
      />
    </div>
    <div
      class="content"
    >
      <div
        class="inline field text"
      >
        <div
          class="ui grid"
        >
          <div
            class="stretched row"
          >
            <div
              class="four wide column"
            >
              <div
                class="wrapper"
              >
                <label
                  for="field-linkTitle"
                >
                  Link Title
                </label>
              </div>
            </div>
            <div
              class="eight wide column"
            >
              <div
                class="ui input"
              >
                <input
                  id="field-linkTitle"
                  name="linkTitle"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="inline field text"
      >
        <div
          class="ui grid"
        >
          <div
            class="stretched row"
          >
            <div
              class="four wide column"
            >
              <div
                class="wrapper"
              >
                <label
                  for="field-linkHref"
                >
                  Link to
                </label>
              </div>
            </div>
            <div
              class="eight wide column"
            >
              <div
                class="ui icon input"
              >
                <input
                  id="field-linkHref"
                  name="linkHref"
                  type="text"
                  value=""
                />
                <i
                  aria-hidden="true"
                  attributes="[object Object]"
                  class="icon"
                />
              </div>
              <button
                class="field-linkHref-action-button"
              >
                <svg
                  class="icon"
                  style="height: 18px; width: auto; fill: currentColor;"
                  viewBox=""
                  xmlns=""
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
